internal/codespaces/api/api.go

```diff
@@ -477,6 +478,73 @@ func (a *API) GetCodespacesMachines(ctx context.Context, repoID int, branch, loc
 	return response.Machines, nil
 }
 
+// TODO - should this be/is this in a more general API?
+func (a *API) GetCodespaceRepoSuggestions(ctx context.Context, partialSearch string) ([]string, error) {
+	reqURL := fmt.Sprintf("%s/search/repositories", a.githubAPI)
+	req, err := http.NewRequest(http.MethodGet, reqURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+
+	parts := strings.Split(partialSearch, "/")
+	partsLen := len(parts)
+	if partsLen > 2 {
+		return nil, errors.New("repository name cannot have multiple slashes")
+	}
+
+	var nameSearch string
+	if partsLen == 2 {
+		user := parts[0]
+		repo := parts[1]
+		nameSearch = fmt.Sprintf("%s user:%s", repo, user)
+	} else {
+		/*
+		 * This results in searching for the text within the owner or the name. It's possible to
+		 * do an owner search and then look up some repos for those owners, but that adds a
+		 * good amount of latency to the fetch which slows down showing the suggestions.
+		 */
+		nameSearch = partialSearch
+	}
+
+	queryStr := fmt.Sprintf("%s in:name", nameSearch)
+
+	q := req.URL.Query()
+	q.Add("q", queryStr)
+	q.Add("sort", "stars")
+	q.Add("per_page", "7") // The prompt shows 7 items so 7 effectively turns off scrolling which is similar behavior to other clients
```

Comment:
Instead of hardcoding this, could it be passed as argument to this method? That way it will be obvious at the call point that only 7 items will be fetched, and if the caller wants to increase this in the future (for example, if we also change the associated Survey prompt configuration) then they can do it easily

Reply:
I like this suggestion - hardcoding `7` made this method a bit too aware that it was used within the Survey results. Refactored it and `sort` out into an optional parameter bundle